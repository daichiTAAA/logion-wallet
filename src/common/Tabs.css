.Tabs .nav-tabs {
    border-bottom: none;
}

.Tabs .nav-tabs .nav-link {
    border-top-left-radius: 10px;
    border-top-right-radius: 0;
    padding: 10px;
    padding-right: 20px;
    min-width: 140px;
    text-align: center;
}

.Tabs .nav-tabs .nav-link:nth-child(n+2) {
    border-top-left-radius: 0;
    margin-left: -10px;
}

.Tabs .nav-tabs .nav-link:last-child {
    border-top-right-radius: 10px;
    padding-right: 10px;
}

.Tabs .nav-tabs .nav-link.active {
    font-weight: bold;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
    padding-right: 10px;
    z-index: 1;
}

.Tabs .nav-tabs .nav-link.active + .nav-link {
    padding-left: 20px;
}

.Tabs .tab-content {
    padding: 20px;
    padding-bottom: 30px;
    border-radius: 10px;
    border-top-left-radius: 0;
}
