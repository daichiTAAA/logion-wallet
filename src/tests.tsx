import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
