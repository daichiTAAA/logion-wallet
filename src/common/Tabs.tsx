import React from 'react';
import ReactTabs from 'react-bootstrap/Tabs';
import ReactTab from 'react-bootstrap/Tab';

import { useCommonContext } from './CommonContext';
import { Children } from './types/Helpers';
import './Tabs.css';

export interface Tab {
    key: string,
    title: Children,
    render: () => Children,
}

export interface Props {
    activeKey: string,
    onSelect: (key: string) => void,
    tabs: Tab[],
}

export default function Tabs(props: Props) {
    const { colorTheme } = useCommonContext();

    const customCss = `
    .Tabs .nav-tabs .nav-link {
        border-color: ${colorTheme.tabs.borderColor};
        color: ${colorTheme.tabs.inactiveTab.foreground};
        background-color: ${colorTheme.tabs.inactiveTab.background};
    }
    .Tabs .nav-tabs .nav-link.active {
        color: ${colorTheme.tabs.foreground};
        background-color: ${colorTheme.tabs.background};
        border-bottom-color: ${colorTheme.tabs.background};
    }
    .Tabs .tab-content {
        color: ${colorTheme.tabs.foreground};
        background-color: ${colorTheme.tabs.background};
        border: 1px solid ${colorTheme.tabs.borderColor};
    }
    `;

    return (
        <div className="Tabs">
            <style>
            { customCss }
            </style>
            <ReactTabs
                activeKey={ props.activeKey }
                onSelect={ key => props.onSelect(key || "") }
            >
                {
                    props.tabs.map(tab => (
                        <ReactTab
                            key={ tab.key }
                            eventKey={ tab.key }
                            title={ tab.title }
                        >
                            { tab.render() }
                        </ReactTab>
                    ))
                }
            </ReactTabs>
        </div>
    );
}
