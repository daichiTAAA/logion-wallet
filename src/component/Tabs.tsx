import React from 'react';
import ReactTabs from 'react-bootstrap/Tabs';
import ReactTab from 'react-bootstrap/Tab';

import { TabsColors } from './ColorTheme';
import { Children } from './types/Helpers';
import './Tabs.css';

export interface Tab {
    key: string,
    title: Children,
    render: () => Children,
}

export interface Props {
    activeKey: string,
    onSelect: (key: string) => void,
    tabs: Tab[],
    colors: TabsColors,
}

export default function Tabs(props: Props) {

    const customCss = `
    .Tabs .nav-tabs .nav-link {
        border-color: ${props.colors.borderColor};
        color: ${props.colors.inactiveTab.foreground};
        background-color: ${props.colors.inactiveTab.background};
    }
    .Tabs .nav-tabs .nav-link.active {
        color: ${props.colors.foreground};
        background-color: ${props.colors.background};
        border-bottom-color: ${props.colors.background};
    }
    .Tabs .tab-content {
        color: ${props.colors.foreground};
        background-color: ${props.colors.background};
        border: 1px solid ${props.colors.borderColor};
    }
    `;

    return (
        <div className="Tabs">
            <style>
            { customCss }
            </style>
            <ReactTabs
                activeKey={ props.activeKey }
                onSelect={ key => props.onSelect(key || "") }
            >
                {
                    props.tabs.map(tab => (
                        <ReactTab
                            key={ tab.key }
                            eventKey={ tab.key }
                            title={ tab.title }
                        >
                            { tab.render() }
                        </ReactTab>
                    ))
                }
            </ReactTabs>
        </div>
    );
}
