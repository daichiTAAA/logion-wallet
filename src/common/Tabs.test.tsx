import { shallowRender } from '../tests';

import Tabs from './Tabs';

test("renders", () => {
    const activeKey = "key1";
    const onSelect = jest.fn();
    const tabs = [
        {
            key: "key1",
            title: "Tab 1",
            render: () => "Content 1"
        },
        {
            key: "key2",
            title: "Tab 2",
            render: () => "Content 2"
        }
    ];
    const result = shallowRender(
        <Tabs
            activeKey={ activeKey }
            onSelect={ onSelect }
            tabs={ tabs }
        />
    );
    expect(result).toMatchSnapshot();
});
