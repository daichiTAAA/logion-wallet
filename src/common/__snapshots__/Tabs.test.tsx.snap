// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Tabs"
>
  <style>
    
    .Tabs .nav-tabs .nav-link {
        border-color: #3b6cf4;
        color: #ffffff;
        background-color: #d8e2fd;
    }
    .Tabs .nav-tabs .nav-link.active {
        color: #ffffff;
        background-color: #0c163d;
        border-bottom-color: #0c163d;
    }
    .Tabs .tab-content {
        color: #ffffff;
        background-color: #0c163d;
        border: 1px solid #3b6cf4;
    }
    
  </style>
  <Tabs
    activeKey="key1"
    mountOnEnter={false}
    onSelect={[Function]}
    unmountOnExit={false}
    variant="tabs"
  >
    <Tab
      eventKey="key1"
      title="Tab 1"
    >
      Content 1
    </Tab>
    <Tab
      eventKey="key2"
      title="Tab 2"
    >
      Content 2
    </Tab>
  </Tabs>
</div>
`;
